src/shogun/evaluation/ClusteringEvaluation.cpp

```diff
@@ -33,17 +52,18 @@ int32_t CClusteringEvaluation::find_mismatch_count(SGVector<int32_t> l1, int32_t
 	return l1.vlen - find_match_count(l1, m1, l2, m2);
 }
 
-void CClusteringEvaluation::best_map(CLabels* predicted, CLabels* ground_truth)
+CLabels* CClusteringEvaluation::best_map(CLabels* predicted, CLabels* ground_truth)
 {
 	ASSERT(predicted->get_num_labels() == ground_truth->get_num_labels())
-	ASSERT(predicted->get_label_type() == LT_MULTICLASS)
-	ASSERT(ground_truth->get_label_type() == LT_MULTICLASS)
+	
+	auto predicted_multiclass = multiclass_labels(predicted);
+	auto ground_truth_multiclass = multiclass_labels(ground_truth);
```

Comment:
not completely related, but it caused me problems when I tried the binary data `classifier_binary_2d_nonlinear_features_train.dat`

Reply:
you needed to add this or this change didnt work?

Reply:
it worked, to interpret binary labels as multiclass, but I don't need it now. thought I leave it anyways since it's better.

Reply:
Pls remove if not needed then We’ll  have to refactor less later

Reply:
@karlnapf when you say `feats = features(something)` it interprets feats as BinaryLabels if possible. If it did, ClusteringEvaluation wouldn't work. This solves this since ClusteringEvaluation should work with both, and since the user can't/shouldn't specify that features should be interpreted as multiclass.

Reply:
ideally, all those `multiclass_labels` calls would be removed and be replaced by assertions (since this system is faulty). As said I have something better in a PR but it is not ready yet. No need to spread the old stuff if it is not really necessary (which isnt the case here).